gwctl/pkg/resourcediscovery/discoverer.go

```diff
@@ -286,6 +285,10 @@ func fetchGatewayClasses(ctx context.Context, k8sClients *common.K8sClients, fil
 			return []gatewayv1.GatewayClass{}, err
 		}
 
+		// because api-server doesn't return TypeMeta in `gatewayClass`
```

Comment:
nit: kube-apiserver does infact return the TypeMeta (it always does)...this mishap happens when that object is decoded. We can likely add https://github.com/kubernetes/kubernetes/issues/80609 as reference.

Reply:
ohh yeah of course it does, my bad (else kubectl-like clients would never show type meta lol), I meant client-go I guess haha.

Anyway, let me change this comment to a ref to the above issue instead.

Thanks for pointing it out!